### PR TITLE
EXIF deletion when importing files

### PR DESCRIPTION
**Pull request by djordjemarusic**
T-And-1167 - Implement EXIF deletion when importing files

### CODE REVIEW

File: mobile/src/main/java/rs/readahead/washington/mobile/media/MediaFileHandler.java [hunk 1]
```diff
@@ -708,4 +717,12 @@ public InputStream getThumbnailStream(final VaultFile vaultFile) {
 
         return null;
     }
+
+    private static void copyStream(InputStream source, OutputStream destination) throws IOException {
+        byte[] buf = new byte[8192];
+        int length;
+        while ((length = source.read(buf)) != -1) {
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/media/MediaFileHandler.java:724]: can we rename `length` to `bytesRead` to make it more descriptive?

File: mobile/src/main/java/rs/readahead/washington/mobile/views/settings/SecuritySettings.kt [hunk 2]
```diff
@@ -104,6 +104,12 @@ class SecuritySettings : BaseFragment() {
             Preferences.setSecurityScreenEnabled(isChecked)
         }
 
+        val keepExifTellaSwitch = binding.keepExifSwitch
+        keepExifTellaSwitch.mSwitch.isChecked = Preferences.isKeepExif()
+        keepExifTellaSwitch.mSwitch.setOnCheckedChangeListener { _: CompoundButton?, isChecked: Boolean ->
+            Preferences.setKeepExif(isChecked)
+        }
+
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/settings/SecuritySettings.kt:112]: what do you think about using apply scope function here ?
```
keepExifTellaSwitch.mSwitch.apply {
    isChecked = Preferences.isKeepExif()
    setOnCheckedChangeListener { _, isChecked ->
        Preferences.setKeepExif(isChecked)
    }
}
```

File: mobile/src/main/java/rs/readahead/washington/mobile/media/MediaFileHandler.java [hunk 3]
```diff
@@ -199,16 +200,24 @@ public static void exportMediaFile(Context context, VaultFile vaultFile, @Nullab
 
     public static Single<VaultFile> importPhotoUri(Context context, Uri uri, @Nullable String parentId) throws Exception {
         // Vault replacement methods
-        InputStream v_input = context.getContentResolver().openInputStream(uri); // original photo
-        Bitmap v_bm = modifyOrientation(BitmapFactory.decodeStream(v_input), v_input); // bitmap of photo
-        Bitmap v_thumb = ThumbnailUtils.extractThumbnail(v_bm, v_bm.getWidth() / 10, v_bm.getHeight() / 10); // bitmap of thumb
-
+        Boolean keepExif = Preferences.isKeepExif();
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/media/MediaFileHandler.java:203]: Let's  change `Boolean` to `boolean` for the keepExif variable, as it is a primitive type.

File: mobile/src/main/java/rs/readahead/washington/mobile/media/MediaFileHandler.java [hunk 4]
```diff
@@ -199,16 +200,24 @@ public static void exportMediaFile(Context context, VaultFile vaultFile, @Nullab
...
+        ByteArrayOutputStream v_image_jpeg_stream = new ByteArrayOutputStream();
         ByteArrayOutputStream v_thumb_jpeg_stream = new ByteArrayOutputStream();
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/media/MediaFileHandler.java:205]: what do you think about renaming `v_image_jpeg_stream` to `imageJpegStream` and `v_thumb_jpeg_stream` to `thumbJpegStream` to follow Java naming conventions and improve readability?

File: mobile/src/main/java/rs/readahead/washington/mobile/media/MediaFileHandler.java [hunk 5]
```diff
@@ -199,16 +200,24 @@ public static void exportMediaFile(Context context, VaultFile vaultFile, @Nullab
...
+
+        InputStream inputStream = context.getContentResolver().openInputStream(uri); // original photo
+        Bitmap v_bm = modifyOrientation(BitmapFactory.decodeStream(inputStream), inputStream); // bitmap of photo
+        Bitmap v_thumb = ThumbnailUtils.extractThumbnail(v_bm, v_bm.getWidth() / 10, v_bm.getHeight() / 10); // bitmap of thumb
         v_thumb.compress(Bitmap.CompressFormat.JPEG, 100, v_thumb_jpeg_stream);
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/media/MediaFileHandler.java:210]: - Let's use the try-with-resources statement to automatically close the input streams (inputStream and input) after use, reducing the need for explicit close calls.
- Combine the declaration and assignment of the inputStream variable within the try block for better scoping and resource handling.
- Reordered the code to improve readability and logical flow.
- Utilized try-with-resources for the input stream within the keepExif branch to automatically close the stream after use.

 ```
    boolean keepExif = Preferences.isKeepExif();
    ByteArrayOutputStream imageJpegStream = new ByteArrayOutputStream();
    ByteArrayOutputStream thumbJpegStream = new ByteArrayOutputStream();
 try (InputStream inputStream = context.getContentResolver().openInputStream(uri)) {
        Bitmap originalBitmap = modifyOrientation(BitmapFactory.decodeStream(inputStream), inputStream); // bitmap of photo
        Bitmap thumbnail = ThumbnailUtils.extractThumbnail(originalBitmap, originalBitmap.getWidth() / 10, originalBitmap.getHeight() / 10); // bitmap of thumb
        thumbnail.compress(Bitmap.CompressFormat.JPEG, 100, thumbJpegStream);

        if (keepExif) {
            try (InputStream input = context.getContentResolver().openInputStream(uri)) {
                copyStream(input, imageJpegStream);
            }
        } else {
            if (!originalBitmap.compress(Bitmap.CompressFormat.JPEG, 100, imageJpegStream)) {
                throw new Exception("JPEG compression failed");
            }
        }
    }
```

**Review by ahlem-jarrar — CHANGES_REQUESTED** [review]

**Review by wafalakhal — APPROVED** [review]

**Pull request merged**